### ISSUE TITLE
Incorrect name specified in <fileType>, should be PSV, actual FileType_PSV

### ISSUE THREAD

**Issue by zolotov** [SeeSharpSoft/intellij-csv-validator#146]
```
java.lang.Throwable: Incorrect name specified in <fileType>, should be PSV, actual FileType_PSV
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:145)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.instantiateFileTypeBean(FileTypeManagerImpl.java:459)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.instantiatePendingFileTypes(FileTypeManagerImpl.java:436)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getRegisteredFileTypes(FileTypeManagerImpl.java:1206)
	at com.intellij.psi.impl.cache.impl.id.IdIndexImpl.getVersion(IdIndexImpl.java:25)
	at com.intellij.util.indexing.FileBasedIndexImpl.registerIndexer(FileBasedIndexImpl.java:336)
	at com.intellij.util.indexing.FileBasedIndexImpl.access$5400(FileBasedIndexImpl.java:98)
	at com.intellij.util.indexing.FileBasedIndexImpl$FileIndexDataInitialization.lambda$initAssociatedDataForExtensions$0(FileBasedIndexImpl.java:2420)
	at com.intellij.util.indexing.IndexInfrastructure$DataInitialization.executeNestedInitializationTask(IndexInfrastructure.java:217)
	at com.intellij.util.indexing.IndexInfrastructure$DataInitialization.runParallelNestedInitializationTasks(IndexInfrastructure.java:205)
	at com.intellij.util.indexing.IndexInfrastructure$DataInitialization.call(IndexInfrastructure.java:166)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:207)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.lambda$run$0(BoundedTaskExecutor.java:185)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:208)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:181)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by SeeSharpSoft** [comment]
@zolotov Thanks for reporting - can you please add IDE type and version? I assume this error occurs on startup, is that correct? The [CI builds and tests](https://travis-ci.org/SeeSharpSoft/intellij-csv-validator) are passing with the latest EAP. So any additional information is very welcome.

**Comment by zolotov** [comment]
Any type, you have wrong file type names for all defined file types in plugin.xml

**Comment by SeeSharpSoft** [comment]
Ok, let me rephrase my question: Since which version does this error occur? I have CI tests covering all versions from 2016.3.2 to the latest EAP - this error doesn't occur here. So I assume it is the latest source code build?

However, we have a tricky situation here - this is how I read the current state with the extension points in the `plugin.xml`:

- the latest EAP (I didn't investigate when exactly this change happened) handles the `fileTypeFactory` extension point slightly different than older version (with respect to `FileTypes` classes that are inherited), but makes correct use of the just recently introduced `fileType` extension point
- in older version (2019.1 and before) the `fileType` extension point does not exist yet, the  `fileTypeFactory` extension point is used

Therefore: If the plugin shall support all version from 2016.3.2 up to the latest EAP,  the extension point `fileTypeFactory` must be used (for versions <=2019.1) as well as the extension point `fileType` for newest version (guess >=2019.2) (I made a comment in the `plugin.xml` regarding "backward compatibility" cause of that). However, if the same name is defined via `fileType` extension as it is defined in the `*FileType` classes - which are used via the `FileTypeFactory` classes - the filetypes are not registered properly (versions >=2019.2) again.

**Summary:** I cannot remove the `fileTypeFactory` extension points for backward compatibility, but I cannt use the `fileTypeFactory` extension point either cause the base-filetype `CsvFileType` is not registered properly in newever version via this extension point. Do I miss something? What I am supposed to do?

As I noticed you are member of Jetbrains, you might be able to clarify this. I can also open a ticket, but as mentioned before: Until now I didn't notice the posted exception.

**Comment by SeeSharpSoft** [comment]
The issue seems to be resolved in upcoming releases. 2019.2.1/2019.2.2 are not working though (https://travis-ci.org/SeeSharpSoft/intellij-csv-validator/builds/584540381). I hope that the fix will be downported to the upcoming release of 2019.2.* to minimize maintenance with respect to version support.

The latest release 2.7.0 is still using the workaround configuration that will fail for newer version, therefore the until build is set to 192.6603.28. I hope to get clarification on how to proceed soon...

**Comment by enerjazzer** [comment]
All the latest CLion EAPs show this error on startup

**Comment by zolotov** [comment]
> Ok, let me rephrase my question: Since which version does this error occur?

@SeeSharpSoft since `fileType` EP was introduced (2019.2?) and you have added `fileType` extension with the wrong language name. There is no tricky situation in fixing this particular exception and its message contains the actual fix to be done:  `Incorrect name specified in <fileType>, should be PSV, actual FileType_PSV`. Other questions should be addressed in the issues you mentioned.

**Comment by SeeSharpSoft** [comment]
> There is no tricky situation in fixing this particular exception

@zolotov Am I getting trolled here?? Please either provide a fix for 2019.2.* or read (and understand) the issue properly!

**Comment by zolotov** [comment]
@SeeSharpSoft I'm trying to help you to make your plugin work, it's kind of weird to yell at me and demand a fix. This issue is about your plugin and exceptions that users get because of it. You can fix it and as I said, there is no tricky question: fix the exception as it's described in the exception message and set proper since/until build number (this is what they were invented for).

Other questions like returning the old behavior of file type factory or other parts IntelliJ platform can be discussed in the IntelliJ issue tracker, I won't participate in these discussions.

**Comment by SeeSharpSoft** [comment]
@zolotov I am sorry for being upset, but if you would have at least read what I wrote in my previous comments and had a look in the linked issues, you would know that `there is no tricky situation` is plain wrong and not helpful at all:

For 2019.2.* using `<fileType>` or using `<fileTypeFactory>` solely or in combination with the correct fileType name is working - it is a bug in IntelliJ and resolved in 2019.3.

I am well aware that the current version does not work for 2019.3 and I am well aware of how to fix it. This issue is blocked as there is nothing I can do for 2019.2.* so far (as of today the most recent available version) and I am currently validating options (see "Validation" channel [here](https://plugins.jetbrains.com/plugin/10037-csv-plugin/versions)) to keep this plugin compatible with all supported versions.

**Comment by zolotov** [comment]
Got it, sorry for being slow. I'm still not fully understand why it doesn't work for 2019.2.*, but I hope the assignee from tickets you mentioned with figure it out.

Do you know the change/issue number that fixed the problem in 2019.3? Maybe I could help with backporting it to 2019.2.5 (it's unlikely to put it in 2019.2.4 already).
